File: Kotlin/AlgorithmsPractice/src/displayLikes/DisplayLikes.kt
package displayLikes

class DisplayLikes(val users: Array<String>) {

    fun getLikesMessage(): String {
        val numberOfLikes: Int = users.size
        var likesMessage: String = "like this"
        if (numberOfLikes != 0) {
            if (numberOfLikes == 1) {
                likesMessage = users[0] + " likes " + likesMessage.split(" ")[1]
            } else if (numberOfLikes == 2) {
                likesMessage = users[0] + " and " + users[1] + " " + likesMessage
            } else if (numberOfLikes == 3) {
                likesMessage = users[0] + ", " + users[1] + " and " + users[2] + " " + likesMessage
            } else {
                likesMessage = users[0] + ", " + users[1] + " and " + (numberOfLikes - 2) + " others " + likesMessage
            }

        } else {
            likesMessage = "Nobody likes " + likesMessage.split(" ")[1]
        }
        return likesMessage
    }
}

File: Python/AlgorithmPractice/display_likes/display_likes.py

class DisplayLikes:
    def __init__(self, users):
        self.users = users

    def get_message(self):
        likes_message = "like this"
        if len(self.users) != 0:
            if len(self.users) == 1:
                likes_message = self.users[0] + " likes " + likes_message.split(" ")[1]
            elif len(self.users) == 2:
                likes_message = self.users[0] + " and " + self.users[1] + " " + likes_message
            elif len(self.users) == 3:
                likes_message = (
                        self.users[0] + ", "
                        + self.users[1] + " and "
                        + self.users[2] + " "
                        + likes_message
                )
            else:
                likes_message = f"{self.users[0]}, {self.users[1]} and {len(self.users) - 2} other users {likes_message}"
        else:
            likes_message = "Nobody likes " + likes_message.split(" ")[1]
        return likes_message
